internals/overlord/checkstate/manager_test.go

```diff
@@ -584,3 +612,259 @@ func changeData(c *C, st *state.State, changeID string) map[string]string {
+				Override:  "replace",
+				Period:    plan.OptionalDuration{Value: time.Second},
+				Threshold: 3,
+				Exec:      &plan.ExecCheck{Command: "echo chk1"},
+			},
+			"chk2": {
+				Name:      "chk2",
+				Override:  "replace",
+				Period:    plan.OptionalDuration{Value: time.Second},
+				Threshold: 3,
+				Exec:      &plan.ExecCheck{Command: "echo chk2"},
+				Startup:   plan.CheckStartupDisabled,
+			},
+			"chk3": {
+				Name:      "chk3",
+				Override:  "replace",
+				Period:    plan.OptionalDuration{Value: time.Second},
+				Threshold: 3,
+				Exec:      &plan.ExecCheck{Command: "echo chk3"},
+				Startup:   plan.CheckStartupEnabled,
+			},
+		},
+	}
+	err := s.planMgr.AppendLayer(origLayer, false)
+	c.Assert(err, IsNil)
+	s.manager.PlanChanged(s.planMgr.Plan())
+	waitChecks(c, s.manager, []*checkstate.CheckInfo{
+		{Name: "chk1", Startup: "enabled", Status: "up", Threshold: 3},
+		{Name: "chk2", Startup: "disabled", Status: "inactive", Threshold: 3},
+		{Name: "chk3", Startup: "enabled", Status: "up", Threshold: 3},
+	})
+	checks, err := s.manager.Checks()
+	c.Assert(err, IsNil)
+	var originalChangeIDs []string
+	for _, check := range checks {
+		originalChangeIDs = append(originalChangeIDs, check.ChangeID)
+	}
+
+	changed, err := s.manager.StartChecks([]string{"chk1", "chk2"})
+	waitChecks(c, s.manager, []*checkstate.CheckInfo{
+		{Name: "chk1", Startup: "enabled", Status: "up", Threshold: 3},
+		{Name: "chk2", Startup: "disabled", Status: "up", Threshold: 3},
+		{Name: "chk3", Startup: "enabled", Status: "up", Threshold: 3},
+	})
+	c.Assert(err, IsNil)
+	c.Assert(changed, DeepEquals, []string{"chk2"})
+	checks, err = s.manager.Checks()
+	c.Assert(err, IsNil)
+	// chk1 and chk3 should still have the same change ID, chk2 should have a new one.
+	c.Assert(checks[0].ChangeID, Equals, originalChangeIDs[0])
+	c.Assert(checks[1].ChangeID, Not(Equals), originalChangeIDs[1])
+	c.Assert(checks[2].ChangeID, Equals, originalChangeIDs[2])
+	// chk2's new Change should be a running perform-check.
+	st := s.overlord.State()
+	st.Lock()
+	change := st.Change(checks[1].ChangeID)
+	status := change.Status()
+	st.Unlock()
+	c.Assert(status, Matches, "Do.*")
+	c.Assert(change.Kind(), Equals, "perform-check")
+}
+
+func (s *ManagerSuite) TestStartChecksNotFound(c *C) {
+	origLayer := &plan.Layer{
+		Checks: map[string]*plan.Check{
+			"chk1": {
+				Name:      "chk1",
+				Override:  "replace",
+				Period:    plan.OptionalDuration{Value: time.Second},
+				Threshold: 3,
+				Exec:      &plan.ExecCheck{Command: "echo chk1"},
+			},
+		},
+	}
+	err := s.planMgr.AppendLayer(origLayer, false)
+	c.Assert(err, IsNil)
+	waitChecks(c, s.manager, []*checkstate.CheckInfo{
+		{Name: "chk1", Startup: "enabled", Status: "up", Threshold: 3},
+	})
+
+	changed, err := s.manager.StartChecks([]string{"chk1", "chk2"})
+	_, ok := err.(*checkstate.ChecksNotFound)
+	c.Assert(ok, Equals, true)
+	c.Assert(err.(*checkstate.ChecksNotFound).Names, DeepEquals, []string{"chk2"})
```

Comment:
Nit: you can avoid the second type assertion by assigning the value here:

```suggestion
	notFoundErr, ok := err.(*checkstate.ChecksNotFound)
	c.Assert(ok, Equals, true)
	c.Assert(notFoundErr.Names, DeepEquals, []string{"chk2"})
```

Reply:
In fact, it's probably better to use [`errors.As`](https://pkg.go.dev/errors#As) these days, which also detects errors which "wrap" `ChecksNotFound`:

```go
var notFoundErr *checkstate.ChecksNotFound
c.Assert(errors.As(err, &notFoundErr), Equals, true)
c.Assert(notFoundErr.Names, DeepEquals, []string{"chk2"})
```